### ISSUE TITLE
`meta: %{"recorded" => true}` not properly populated, leading to jobs without results?

### ISSUE THREAD

**Issue by dballance** [sorentwo/oban#648]
### Environment

- Oban Version 
```
        {:oban, "~> 2.11"},
        {:oban_pro, "~> 0.10", repo: "oban"},
        {:oban_web, "~> 2.9", repo: "oban"},
```
- PostgreSQL Version - `13.2` - running a local container
- Elixir & Erlang/OTP Versions (`elixir --version`)
```
Erlang/OTP 24 [erts-12.2.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]
Elixir 1.13.2 (compiled with Erlang/OTP 22)
```
- Oban Config
```
# :core is a local module that spans our repo module and web module. It's where orchestration bits go.
config :core, Oban,
  prefix: "jobs",
  repo: MyApp.Repo,
  engine: Oban.Pro.Queue.SmartEngine,
  notifier: Oban.Notifiers.PG,
  plugins: [
    Oban.Plugins.Gossip,
    Oban.Web.Plugins.Stats,
    Oban.Pro.Plugins.DynamicPruner,
    Oban.Pro.Plugins.DynamicLifeline
  ],
  queues: [
    default: [local_limit: 3, global_limit: 10]
  ]
```

### Current Behavior

When I execute a job that is recorded - it *sometimes* is recorded. Typically, if it works, it will work until I restart the beam session (started via `iex -S mix phx.server`). If it doesn't work, no jobs have the `:meta` key properly populated with `"recorded"` value. Until I restart the session, no jobs properly start recorded.

If I restart, and a the job returns properly from `new()`, then it tends to stay that way. 

I can consistently replicate by `mix deps.clean` on `oban` and `oban_pro` and then `iex -S mix phx.server`. First start everything works appropriately, second start the `meta` key is missing from the changeset.

This could totally be my environment - I'm working to integrate into a deployed cluster environment, but running locally in a single node environment to learn and play with Oban.

```elixir
defmodule MyApp.Jobs.Recorded do
  use Oban.Pro.Worker, recorded: true

  @impl true
  def process(%Job{} = job) do
    IO.puts("Executing job - #{job.id}")
    value = :rand.uniform(255)
    {:ok, value}
  end
end
```

Run below and observe that meta may, or may not, be populated
```
iex(1)> %{} |> MyApp.Jobs.Recorded.new()
```
### Expected Behavior

Setting recorded on a job always sets the proper meta values to ensure job results are recorded.

**Comment by dballance** [comment]
It also appears that manually setting `meta: %{"recorded" => true}` in the options of `new` does not work around this issue. The appropriate `pre` and `post` hooks are not added, and in `oban_web` the job shows as "recorded" but no recordings found.

**Comment by sorentwo** [comment]
@dballance The good news is that I’m fairly confident that is a dev specific issue. The bad news is that I need to debug and figure out why it loses the recording stage sometimes (seemingly after a reload).

**Comment by sorentwo** [comment]
This was a strange one. I was eventually able to reproduce it and put a fix together. Watch for the fix in Pro v0.10.2.

**Comment by dballance** [comment]
Thanks!
